src/SFA.DAS.EAS.Portal.UnitTests/Application/Commands/Cohort/CohortApprovalRequestedCommandHandlerTests.cs

```diff
@@ -0,0 +1,220 @@
+﻿using NUnit.Framework;
+using SFA.DAS.Commitments.Api.Client.Interfaces;
+using SFA.DAS.Commitments.Api.Types.Commitment;
+using SFA.DAS.EAS.Portal.Application.Commands.Cohort;
+using SFA.DAS.EAS.Portal.UnitTests.Builders;
+using System.Threading.Tasks;
+using FluentAssertions;
+using System.Linq;
+using System;
+using System.Collections.Generic;
+using SFA.DAS.EAS.Portal.Application.Services;
+using System.Threading;
+using SFA.DAS.EAS.Portal.Types;
+using Moq;
+using SFA.DAS.EAS.Portal.Client.Database.Models;
+
+namespace SFA.DAS.EAS.Portal.UnitTests.Application.Commands.Cohort
+{
+    [Parallelizable]
+    [TestFixture]
+    public class CohortApprovalRequestedCommandHandlerTests
+    {
+        public class TestContext
+        {
+            public CohortApprovalRequestedCommandHandler Sut { get; private set; }
+            public Account TestAccount { get; private set; }
+            public AccountDocument TestAccountDocument { get; private set; }
+            public CommitmentView TestCommitment { get; private set; }
+            public Mock<IAccountDocumentService> MockAccountsService { get; private set; }
+            public Mock<IProviderCommitmentsApi> MockProviderCommitmentsApi { get; private set; }
+
+            public TestContext()
+            {
+                TestAccount = new AccountBuilder().WithOrganisation(new OrganisationBuilder());
+                TestAccountDocument = new AccountDocument() { Account = TestAccount };
+                TestCommitment = new CommitmentViewBuilder();
+
+                MockAccountsService = new Mock<IAccountDocumentService>();
+
+
+                MockAccountsService
+                    .Setup(m => m.Get(It.IsAny<long>(), It.IsAny<CancellationToken>()))                    
+                    .ReturnsAsync(TestAccountDocument);
+
+                MockProviderCommitmentsApi = new Mock<IProviderCommitmentsApi>();
+
+                MockProviderCommitmentsApi
+                    .Setup(m => m.GetProviderCommitment(It.IsAny<long>(), It.IsAny<long>()))
+                    .ReturnsAsync(TestCommitment);
+
+                Sut = new CohortApprovalRequestedCommandHandler(MockAccountsService.Object, MockProviderCommitmentsApi.Object);
+            }
+        }
+
+        public class Handle : CohortApprovalRequestedCommandHandlerTests
+        {
+            [Test]
+            public async Task WhenCalled_ThenTheAccountServiceIsCalledToRetrieveTheAccount()
+            {
+                // arrange
+                var testContext = new TestContext();
+                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
+
+                // act
+                await testContext.Sut.Handle(command);
+
+                //assert
+                testContext.MockAccountsService.Verify(m => m.Get(command.AccountId, It.IsAny<CancellationToken>()), Times.Once);
```

Comment:
It's unsafe to use expected values, where the code under test has the opportunity to change the expected value.

E.g. `await testContext.Sut.Handle(command)` could change `AccountId` in the command object, which is then used during the assert phase, and depending on whether `AccountId` was changed before or after the verified call, we could get a false positive or a false negative.

Other tests in this PR do the same thing.